fb.c

```diff
@@ -145,3 +145,290 @@ void destroy_framebuffer(const char *proc_fname)
     pr_debug("Destroying framebuffer %s\n", proc_fname);
     remove_proc_entry(proc_fname, NULL);
 }
+
+static int vcam_fb_check_var(struct fb_var_screeninfo *var,
+                             struct fb_info *info)
+{
+    if (!var->xres)
+        var->xres = 1;
+    if (!var->yres)
+        var->yres = 1;
+    if (var->xres > var->xres_virtual)
+        var->xres_virtual = var->xres;
+    if (var->yres > var->yres_virtual)
+        var->yres_virtual = var->yres;
+
+    var->xres_virtual = var->xoffset + var->xres;
+    var->yres_virtual = var->yoffset + var->yres;
+
+    if (var->bits_per_pixel <= 16)
```

Comment:
I dislike the way you set bpp. Can you use the trick of align up as Linux kernel?
See https://lists.kernelnewbies.org/pipermail/kernelnewbies/2013-November/009229.html